basic_pipeline_extension/01.0_Bin.md

```diff
@@ -0,0 +1,106 @@
+# Bin
+
+A Membrane's bin is a container for elements, which allows for creating reusable groups of elements.
+Bin is similar to a pipeline in that it consists of linked elements. Such bin can then be placed inside a pipeline and linked with other elements. Bins can also be nested within one another.
+Bin also has another advantage - it manages its children, for instance by dynamically spawning or replacing them as the stream changes.
+
+# Enclosing pipeline elements inside a bin
+
+As you can see, we have `Source` -> `Ordering Buffer` -> `Depayloader` chain, which is duplicated.
+![Pipeline scheme](../basic_pipeline/assets/images/basic_pipeline.png) <br>
+
+We can encapsulate these elements inside `Bin`.
+![Pipeline scheme using bin](/basic_pipeline_extension/assets/images/basic_pipeline_bin.png) <br>
+
+Just like with a pipeline, we have to define bin's children and the links between them.
+
+###### **`lib/Bin.ex`**
+```Elixir
+defmodule Basic.Bin do
+  use Membrane.Bin
+
+  def_output_pad :output,
+    demand_unit: :buffers,
+    caps: {Basic.Formats.Frame, encoding: :utf8}
+
+  @impl true
+  def handle_init(_opts) do
+    children = %{
+      input: %Basic.Elements.Source{location: "input.A.txt"},
+      ordering_buffer: Basic.Elements.OrderingBuffer,
+      depayloader: %Basic.Elements.Depayloader{packets_per_frame: 4}
+    }
+
+    links = [
+      link(:input) |> to(:ordering_buffer) |> to(:depayloader) |> to_bin_output(:output)
+    ]
+
+    spec = %ParentSpec{children: children, links: links}
+
+    {{:ok, spec: spec}, %{}}
+  end
+end
+```
+
+The first thing you may notice is that unlike with the pipeline we are defining output pads. This is because the bin has to send its data to `Mixer`.
```

Comment:
```suggestion
The first thing you may notice is that unlike with the pipeline we are defining output pads. This is because we want the bin to send its data to the following element or a bin in the pipeline - in case of our pipeline, to the `Mixer`.
```

Well, bin does not have to send its data to the next element/bin in all cases - i.e. we might have a bin that is finished with a sink.